pulp_deb/app/serializers/content_serializers.py

```diff
@@ -721,13 +728,71 @@ class ReleaseSerializer(NoArtifactContentSerializer):
     A Serializer for Release.
     """
 
+    def get_unique_together_validators(self):
+        """
+        We do not want UniqueTogetherValidator since we have retrieve logic!
+        """
+        return []
+
     codename = CharField()
     suite = CharField()
     distribution = CharField()
     version = NullableCharField(required=False, allow_null=True, default=None)
     origin = NullableCharField(required=False, allow_null=True, default=None)
     label = NullableCharField(required=False, allow_null=True, default=None)
     description = NullableCharField(required=False, allow_null=True, default=None)
+    architectures = ListField(child=CharField(), required=False)
+    components = ListField(child=CharField(), required=False)
+
+    @staticmethod
+    def _get_or_create_content_pk(model, **data):
+        content, created = model.objects.get_or_create(**data)
+        if created:
+            CreatedResource(content_object=content).save()
```

Comment:
```suggestion
        if created:
            CreatedResource(content_object=content).save()
        else:
            content.touch()
```
We may need to protect this from cleanup.